contract_sale_invoicing/models/contract.py

```diff
@@ -9,23 +9,139 @@ class ContractContract(models.Model):
+                                    'product_id'
+                                ) in sale_order_line_product_qty:
+                                    if sale_order_line_product_qty.get(
+                                            line.product_id.id
+                                    ) > invoice_line.get('quantity'):
+                                        remain_qty = \
+                                            sale_order_line_product_qty.\
+                                            get(invoice_line.get('product_id')
+                                                ) - invoice_line.get(
+                                                'quantity') or 0
+                                        invoice_line_list.append({
+                                            'uom_id':
+                                            contract_line_rec.uom_id.id,
+                                            'product_id':
+                                            invoice_line.get('product_id'),
+                                            'quantity': remain_qty or 0,
+                                            'discount':
+                                            contract_line_rec.discount,
+                                            'contract_line_id':
+                                            contract_line_rec.id,
+                                            'name': contract_line_rec.name,
+                                            'account_analytic_id':
+                                            contract_line_rec.
+                                            analytic_account_id.id,
+                                            'sale_line_ids':
+                                            [(6, 0, [line.id])],
+                                            'price_unit':
+                                            contract_line_rec.price_unit,
+                                            'account_id':
+                                            order_id.partner_id.
+                                            property_account_receivable_id.id,
+                                        })
+                                        sale_order_line_product_qty[
+                                            line.product_id.id] -= remain_qty
+                                if line.product_id.id != invoice_line.get(
+                                        'product_id'):
+                                    invoice_line_list.append({
+                                        'uom_id': line.product_uom.id,
+                                        'product_id': line.product_id.id,
+                                        'quantity': line.product_uom_qty or 0,
+                                        'contract_line_id':
+                                        contract_line_rec.id,
+                                        'name': line.name,
+                                        'sale_line_ids': [(6, 0, [line.id])],
```

Comment:
in this case you will link the invoice line only to the first order line with the same product